internal/repos/fs.go

```diff
@@ -54,34 +54,40 @@ func NewFileRepo(config map[string]any, spec model.RepoSpec) (*FileRepo, error)
 	}, nil
 }
 
-func (f *FileRepo) Push(ctx context.Context, id model.TMID, raw []byte) error {
+func (f *FileRepo) Push(ctx context.Context, id model.TMID, raw []byte, opts PushOptions) (PushResult, error) {
```

Comment:
I have the feeling that returning both, a PushResult and an error, makes it a bit error-prone in further handling,
as the current bugs show.

In my option it would be better, only to return the PushResult holding the error within the PushResult struct, or just as a message.
Then it would be more intuitive in further handling
to focus on evaluating the PushResultType  (OK, Warning, Error, Exists)

e.g.

`func (f *FileRepo) Push(ctx context.Context, id model.TMID, raw []byte, opts PushOptions) PushResult`

```
type PushResult struct {
	Type PushResultType
	Err error    // in case of "warning", also an error would be set
	TmID string
}
```
```
or as it currently is:
type PushResult struct {
	Type PushResultType
	Text string   // text can contain error message or warning message
	TmID string
}
```



Maybe also `PushResultTMExists` doesn't have to be its own PushResultType but returned as PushResultType `PushResultError`.
The PushResult field "Err" is then set to "ErrTMIdConflict" and evaluated later (e.g. for print out)

Reply:
You're right. This is, indeed, confusing. I'm going to rework it